src/Framework/Framework/Controls/SelectorBase.cs

```diff
@@ -87,15 +88,51 @@ public bool Enabled
         [ControlUsageValidator]
         public static IEnumerable<ControlUsageError> ValidateUsage(ResolvedControl control)
         {
-            if (control.Properties.ContainsKey(SelectorBase.ItemValueBindingProperty) &&
-                control.Properties.GetValue(SelectorBase.ItemValueBindingProperty) is ResolvedPropertyBinding valueBinding)
+            if (control.GetValue(SelectorBase.ItemValueBindingProperty) is ResolvedPropertyBinding valueBinding)
             {
                 var t = valueBinding.Binding.ResultType;
                 if (!t.IsPrimitiveTypeDescriptor())
                 {
                     yield return new ControlUsageError("Return type of ItemValueBinding has to be a primitive type!", valueBinding.DothtmlNode);
                 }
             }
+
+            if (control.GetValue(SelectorBase.ItemTextBindingProperty) is ResolvedPropertyBinding textBinding)
+            {
+                // TODO: just change the property type to IValueBinding<string>?
+                var t = textBinding.Binding.ResultType;
```

Comment:
This should be resolved so that types, such as `DateTime` are displated correctly